instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/NetClientAttributesExtractor.java

```diff
@@ -49,17 +50,26 @@ public void onEnd(
 
     internalSet(attributes, SemanticAttributes.NET_TRANSPORT, getter.transport(request, response));
 
-    String peerIp = getter.peerIp(request, response);
     String peerName = getter.peerName(request, response);
-
-    if (peerName != null && !peerName.equals(peerIp)) {
-      internalSet(attributes, SemanticAttributes.NET_PEER_NAME, peerName);
-    }
-    internalSet(attributes, SemanticAttributes.NET_PEER_IP, peerIp);
+    String sockPeerAddr = getter.sockPeerAddr(request, response);
 
     Integer peerPort = getter.peerPort(request, response);
-    if (peerPort != null && peerPort > 0) {
-      internalSet(attributes, SemanticAttributes.NET_PEER_PORT, (long) peerPort);
+    Integer sockPeerPort = getter.sockPeerPort(request, response);
+
+    if (peerName != null && !peerName.equals(sockPeerAddr)) {
+      internalSet(attributes, SemanticAttributes.NET_PEER_NAME, peerName);
```

Comment:
@lmolkova I'm wondering now if this logic is right based on rpc.md in your spec PR:

> SHOULD set `net.peer.name` to the IP address provided in the host component

I guess my question is: if we are given a `new InetSocketAddress("127.0.0.1", 1234)`, should we set:

* `net.peer.name` to `127.0.0.1`
* `net.peer.port` to 1234

and leave `net.sock.peer.addr` and `net.sock.peer.port` empty?

or set:

* `net.peer.name` to `127.0.0.1`
* `net.sock.peer.addr` to `127.0.0.1` (since this provides more info)
* `net.peer.port` to 1234